imports/server/accounts.ts

```diff
@@ -45,6 +45,32 @@ const summaryFromLoginInfo = function (info: LoginInfo) {
   }
 };
 
+Accounts.onCreateUser((_, user) => {
+  if (user.services?.google) {
+    // TODO(#2021): Enable this when we have an additional ACL check.
+    throw new Meteor.Error(403, "No access");
+
+    // Pull profile data from the user's Google profile.
+    // user.emails = [{address: user.services.google.email, verified: true}];
+    // user.googleAccount = user.services.google.email;
+    // user.googleAccountId = user.services.google.id;
+    // user.displayName = user.services.google.name;
+  }
+  return user;
+});
+
+Accounts.setAdditionalFindUserOnExternalLogin(
+  (info: Accounts.ExternalLoginInfo) => {
+    if (info.serviceName === "google") {
+      // Attempt to match existing Google users by their account ID.
+      // We can't use the async method since Meteor's API only takes a sync one.
+      // eslint-disable-next-line jolly-roger/no-disallowed-sync-methods
+      return MeteorUsers.findOne({ googleAccountId: info.serviceData.id });
```

Comment:
Oh we probably need to make sure there's only one user with that Google ID, because we don't enforce uniqueness. So, like, find with limit=2 and make sure you get exactly 1 back

Reply:
On digging further, this might be a bit more involved to get right.

Meteor is tracking the Google association in its own field, services.google.id ([ref](https://github.com/meteor/meteor/blob/c26569d224eb081b9aef950f44066c555ca9b6f9/packages/accounts-base/accounts_server.js#L1296)). While the current PR does a one-time link between these two fields, looking for JR users by existing googleAccountId, and (in the commented-out code) populating the googleAccountId from the service ID on new account creation, after this point, nothing is keeping these in sync. So you can still run into weirdness by e.g. making an association with a unique account at the time, and then associating more accounts with that Google account via the JR profile page.

Do you have any opinions on how we should handle this? I feel conceptually like having two different fields for this will lead to pain; maybe we should just migrate our existing fields to the standard Meteor ones, or will that cause other problems?

Reply:
Ugh yes that is annoying. I agree that if we're going to support logging in with Google account, then we should also migrate our existing fields onto the Meteor standard ones.

Here are the reasons I can think of that it's going to suck:
- We now need to care about uniqueness. Unfortunately we *do* actually need to explicitly handle this somehow, at least for the Death and Mayhem deployment. I just checked and we have about 25 Google accounts that are associated with more than one JR account. (Realistically, I'm pretty sure all of these should be merged into a single account, but that's a more complicated problem)
- We probably need to migrate incrementally (as people log in or re-link their accounts), rather than being able to do it all at once. The `googleAccount` and `googleAccountId` fields aren't enough to populate Meteor's service record. I guess it's possible that we can fudge it (and just not have an accessToken or any of the other fields), but I'm not sure if that will break things down the line. My personal preference would be to lazily migrate the data, either when someone logs in using an existing Google account or when they re-link their Google account, and then at some point in the future we can deprecate that and write a database migration to remove any users that still have the old fields set.

Reply:
After thinking about it more, I'm less convinced that it's worth using the standard mechanism at this time. Now that we're moving towards invitation links rather than other Google-based methods, we don't have a particular use case for it right now, and a custom login handler that uses our existing fields nets out to be simpler than the standard one, given that most of the complexity is already in the google-oauth module anyway, and that we don't have to deal with any migration questions at the moment.

So I've pivoted this PR to using a custom login handler instead, and we could punt migrating to the standard handler unless/until there's a more compelling reason for it.

I did find a bug along the way - unlinking a Google Account only cleared the linked email but not the ID. I've fixed that here, but if this PR doesn't get merged for any reason we should make sure not to lose track of that.

Let me know what you think!